File: programacao/c/respostas/operacoes/28.c
#include <stdio.h>
#include <math.h> // Biblioteca para sqrt()

int main() {
    const double R = 6371.0;
    double h, d;

    printf("Digite a altura do observador (em metros): ");
    scanf("%lf", &h);

    h /= 1000.0;

    d = sqrt(2 * h * R + pow(h, 2));

    printf("A distancia ate o horizonte e %.2f km (%.0f metros)\n", d, d * 1000);

    return 0;
}

File: programacao/python/respostas/operacoes/28.py
import math

R = 6371.0
h = float(input("Altura do observador (metros): "))

h = h / 1000.0
d = math.sqrt(2 * h * R + h**2)

print(f"A distancia ate o horizonte e {d:.2f} km ({d * 1000:.0f} metros)")
